libtransmission/torrent-magnet.cc

```diff
@@ -380,15 +386,8 @@ double tr_torrentGetMetadataPercent(tr_torrent const* tor)
         return 1.0;
     }
 
-    if (auto const& m = tor->incomplete_metadata; m)
-    {
-        if (auto const& n = m->piece_count; n != 0)
```

Comment:
Not so sure, @tearfur, but maybe it should be `auto const n` instead of `auto const& n`?
That's the only "difference" I'm seeing apart from the std:unique_ptr adoption.

Reply:
Unless another thread can modify `m->piece_count` (there isn't btw), I don't see how this can make a difference. But you are right that `auto const n` is better than `auto const& n` here.

My best guess would be a buggy `std::unique_ptr` implementation in older Xcode (thus older `libc++`). It is not as mature as libstdc++, and libtransmission has been affected by their bugs before.

Reply:
I've addressed `auto const& n` in #6311, but no improvements.
And this whole rollback still does show seemingly better results at fetching magnet metadata.
So maybe you're right, an issue in `std::unique_ptr`? And maybe that could explain potential other regressions in download performances in past two years?

Reply:
Whilst `std::unique_ptr` is the only explanation we have now, it is hard to imagine something like this hasn't been caught... 

I don't think the way I used `std::unique_ptr` here is unusual...?